plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java

```diff
@@ -75,7 +153,7 @@ public static <T> ActionListener<T> wrapRestActionListener(ActionListener<T> act
                 String errorMessage = generalErrorMessage;
                 if (isBadRequest(e) || e instanceof MLException) {
                     errorMessage = e.getMessage();
-                } else if (cause != null && (isBadRequest(cause) || cause instanceof MLException)) {
+                } else if (isBadRequest(cause) || cause instanceof MLException) {
```

Comment:
Why remove `cause != null &&` ?

Reply:
@zane-neo can help here

Reply:
`Throwables.getRootCause(e)` never returns null, removing this won't cause any issue.